tests/cursor/jsfmt.spec.js

```diff
@@ -14,3 +14,11 @@ test("positions cursor relative to closest node, not SourceElement", () => {
     cursorOffset: 7
   });
 });
+
+test("keeps cursor inside formatted node", () => {
+  const code = "return         15";
+  expect(prettier.formatWithCursor(code, { cursorOffset: 14 })).toEqual({
+    formatted: "return 15;\n",
+    cursorOffset: 14 // TODO fix this
```

Comment:
Note that the result offset is nevertheless outside the node, since the cursor is one space before the `15`, so we still need something like https://github.com/prettier/prettier/pull/1983 (or maybe some logic that finds the "nearest" node to the cursor, rather than the smallest one containing the cursor).